### PR TITLE
fix(Button menu): button menu accessibility issue - 3519

### PR DESCRIPTION
**Pull request by amal-k-joy**
Contributes to #3519 

Follow a11y guideline about "accessible label should contain or match the visible labels) or allow overrides

#### What did you change?
Provide the menuAriaLabel prop to be passed to the Button menu component. This optional label should contain or match the visible labels or it will automatically take button label

#### How did you test and verify your work?
Verified in the storyboard using IBM Equal Access Accessibility Checker.

### CODE REVIEW

**Review by matthewgallo — CHANGES_REQUESTED** [review]
There appear to be some lint issues, running `yarn format` should resolve this for you.

**Comment by amal-k-joy** [comment]
> There appear to be some lint issues, running `yarn format` should resolve this for you.
@matthewgallo 
I have resolved as advised

**Review by Ratheeshrajan — COMMENTED** [review]
The accessibility checker requires that the visible text/label match the `area-label`, in my opinion. The addition of the new prop `menuArearLabel` could potentially lead to an accessibility violation. Take a look at this example:
1. Maintain the label on the `Primary button` menu button.
2. Add `menuAreaLabel` something different text
3. Restart the accessibility checker.

In my openion we only need to add `areaLabel={label}` in `ButtonMenu` component.  @matthewgallo could you add your thoughts.

**Comment by matthewgallo** [comment]
I do like the simplicity of just using the `label` as the `ariaLabel` however, it can be a node which I don't think is a valid value to pass to `aria-label`. You might consider the approach that @devadula-nandan [made recently in the Tearsheet](https://github.com/carbon-design-system/ibm-products/pull/4130/files#diff-b8bde42986b0f2763c922b3e9bfa4f7deb72e4dcfee6731cb498313f9e4c47ecR218) where if there is an ariaLabel prop passed in (in your case menuAriaLabel) we use that, otherwise we try to set it to the text content of the `label` prop since it could be a node.

File: packages/ibm-products/src/components/ButtonMenu/ButtonMenu.test.js [hunk 1]
```diff
@@ -27,11 +27,11 @@ const icon = Add16;
 const iconDescription = `Icon ${uuidv4()}`;
 const itemText = `Option ${uuidv4()}`;
 const label = `Button ${uuidv4()}`;
-const ariaLabel = `aria ${label} label`;
+const menuAriaLabel = `aria ${label} label`;
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ButtonMenu/ButtonMenu.test.js:30]: This variable menuAriaLabel is used to make it consistent with V2 code base.

**Comment by amal-k-joy** [comment]
> I do like the simplicity of just using the `label` as the `ariaLabel` however, it can be a node which I don't think is a valid value to pass to `aria-label`. You might consider the approach that @devadula-nandan [made recently in the Tearsheet](https://github.com/carbon-design-system/ibm-products/pull/4130/files#diff-b8bde42986b0f2763c922b3e9bfa4f7deb72e4dcfee6731cb498313f9e4c47ecR218) where if there is an ariaLabel prop passed in (in your case menuAriaLabel) we use that, otherwise we try to set it to the text content of the `label` prop since it could be a node.

I have used getNodeTextContent , so that it will either take from menuAriaLabel prop or take the text content of the label.

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**